iothub/device/src/ITransportSettings.cs

```diff
@@ -20,5 +22,41 @@ public interface ITransportSettings
         /// The time to wait for a receive operation.
         /// </summary>
         TimeSpan DefaultReceiveTimeout { get; }
+
+        /// <summary>
+        /// The web proxy that will be used to connect to IoT Hub using a web socket connection for AMQP or MQTT; or when using the HTTP protocol.
+        /// </summary>
+        /// <value>
+        /// An instance of a class that implements <see cref="IWebProxy"/>.
+        /// </value>
+        /// <remarks>
+        /// This setting will be used when the client attempts to connect over web sockets. For example, if the client attempts to connect to IoT hub using <see cref="TransportType.Amqp"/> or <see cref="TransportType.Mqtt"/> the client will first try over TCP. If that fails, the client will fall back to using web sockets and will use the proxy setting. The setting will also be used when <see cref="TransportType.Amqp_WebSocket_Only"/>, <see cref="TransportType.Mqtt_WebSocket_Only"/>, or <see cref="TransportType.Http1"/> is specified. 
+        /// </remarks>
+        /// <example>
+        /// To set a proxy you must instantiate an instance of the <see cref="WebProxy"/> class--or any class that derives from <see cref="IWebProxy"/>. The snippet below shows a method that returns a device using a proxy that connects to localhost on port 8888.
+        /// <code>
+        /// static DeviceClient GetClientWithProxy()
+        /// {
+        ///     try
+        ///     {
+        ///         var proxyHost = "localhost";
+        ///         var proxyPort = 8888;
+        ///         // In the event the AMQP ports are blocked from this device, we will fall back to AMQP over web sockets
+        ///         var transportSettings = new AmqpTransportSettings(Microsoft.Azure.Devices.Client.TransportType.Amqp)
```

Comment:
Shouldn't the best practice be that they specify Amqp over websocket explicitly?

Reply:
Do we have this explicitly defined as a best practice anywhere? I would think that specifying the specific protocol is good for ensuring you choose one over the other, but I think we've just left it up to the customer.